src/lib.rs

```diff
@@ -29,15 +30,59 @@ mod dlmalloc;
 #[cfg(all(feature = "global", not(test)))]
 mod global;
 
+/// A platform interface
+pub trait System: Send {
+    /// Allocates a memory region of `size` bytes
```

Comment:
Could this be expanded? I think that `size` is expected to be a multiple of the page size? I  think this should also mention that this is expected to allocate from the system.

Reply:
Also can the return value be documented?

Reply:
(can the return values of all these functions be documented?)

Reply:
I've added more documentation. `size` in `System::alloc` isn't guaranteed to be a multiple of page size. It only seems to work out nicely for the current platforms as `DEFAULT_GRANULARITY` is a multiple of their page sizes. [call cite](https://github.com/alexcrichton/dlmalloc-rs/blob/master/src/dlmalloc.rs#L355)